### PR TITLE
Add ippool CRD

### PR DESCRIPTION
**Pull request by zhengxiexie**

### CODE REVIEW

File: pkg/apis/v1alpha2/ippool_types.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
```
Comment (lxiaopei) [inline pkg/apis/v1alpha2/ippool_types.go:1]: 2022 -> 2023

File: pkg/apis/v1alpha2/ippool_types.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// IPPool is the Schema for the ippools API
```
Comment (lxiaopei) [inline pkg/apis/v1alpha2/ippool_types.go:15]: add period '.' at the end.

File: pkg/apis/v1alpha2/ippool_types.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPPoolSpec   `json:"spec"`
+	Status IPPoolStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IPPoolList contains a list of IPPool
```
Comment (lxiaopei) [inline pkg/apis/v1alpha2/ippool_types.go:26]: ditto.

File: pkg/apis/v1alpha2/groupversion_info.go
Comment (lxiaopei) [inline pkg/apis/v1alpha2/groupversion_info.go:1]: did you sync to the latest code? The vpc_dev branch latest code should already include this file.

File: pkg/apis/v1alpha2/zz_generated.deepcopy.go
Comment (lxiaopei) [inline pkg/apis/v1alpha2/zz_generated.deepcopy.go:1]: did you sync to the latest code? The vpc_dev branch latest code should already include this file

**Comment by lxiaopei** [comment]
Could you check the e2e failure?

File: pkg/apis/v1alpha2/groupversion_info.go
Comment (zhengxiexie) [reply pkg/apis/v1alpha2/groupversion_info.go:1]: vpc_dev is v1alpha1.

File: pkg/apis/v1alpha2/zz_generated.deepcopy.go
Comment (zhengxiexie) [reply pkg/apis/v1alpha2/zz_generated.deepcopy.go:1]: ditto.

**Comment by zhengxiexie** [comment]
> Could you check the e2e failure?

We need new testbeds.

**Review by dantingl — APPROVED** [review]

**Pull request merged**